### PR TITLE
product page initial commit

### PR DESCRIPTION
**Pull request by anshulraheja**
Product Page with Filter functionality only.

Files to review
1. ProductCard.js
2. DbContext.js
3. Filter-context.js
4. FilterBar
5. ProductPage

### CODE REVIEW

File: src/components/Header/Header.css [hunk 1]
```diff
@@ -1,15 +1,11 @@
...
-    top: 0; */
     display: flex;
     align-items: center;
     justify-content: space-between;
     gap: 0.5rem;
     z-index: 1;
+    margin-bottom: 2rem;
 }
 .header a {
     color: black;
```
Comment (parulgupta2001) [inline src/components/Header/Header.css:11]: use var

**Review by parulgupta2001 — APPROVED** [review]
Apply the change which I suggested, otherwise everything looks good.

File: src/backend/db/products.js [hunk 2]
```diff
@@ -12,252 +12,53 @@ export const products = [
...
-    categoryName: "Women",
+    subCategory: "Shoes",
+    rating: "5"
   },
   {
     _id: uuid(),
     title: "Nike3",
     price: "1000",
-    image: "https://static.nike.com/a/images/t_PDP_1728_v1/f_auto,b_rgb:f5f5f5/b7d9211c-26e7-431a-ac24-b0540fb3c00f/air-force-1-07-shoe-WrLlWX.png",
+    image: "https://static.nike.com/a/images/t_PDP_1728_v1/f_auto,b_rgb:f5f5f5/3e797e14-9abe-44b7-85c7-d25f5751a2d3/pro-older-short-sleeve-training-top-1z0zqt.png",
```
Comment (PuneetP16) [inline src/backend/db/products.js:58]: Suggestion: You can store all the images in one object and then extract the image url in image property

example:
const imgObj = {
    nike3: "/image/url/"
}

you could it as:
image: imgObj.nike3

File: src/components/FilterBar/FilterBar.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                            return (
+                                <div key={_id}>
+                                    <label htmlFor={_id} >
+                                        <input type="checkbox" id={_id} name="category"
+                                            onChange={e =>
+                                                filterDispatch({
+                                                    type: subCategoryName.toUpperCase(),
+                                                })
+                                            }
+                                        // checked={subCategoryName.toLowerCase()}
```
Comment (PuneetP16) [inline src/components/FilterBar/FilterBar.js:41]: remove comments

File: src/components/FilterBar/FilterBar.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                    </label>
+                    <label htmlFor="highToLow">
+                        <input type="radio" id="highToLow" name="sortby" value="highToLow"
+                            onChange={() =>
+                                filterDispatch({
+                                    type: "SORT_BY_PRICE",
+                                    payload: "HIGH_TO_LOW"
+                                })
+                            }
+                            checked={sortBy === "HIGH_TO_LOW" ? true : false}
```
Comment (PuneetP16) [inline src/components/FilterBar/FilterBar.js]: We don't need to use the ternary operator here.
sortBy === "HIGH_TO_LOW"  will return either true or false by default.

**Review by PuneetP16 — APPROVED** [review]
Left few suggestions to check.

Must fix: Clear all button is not resetting the checked state of categories
image

Good to merge after implementing necessary changes

**Pull request merged**